### DISCUSSION TITLE
How can I create a tweet with medias using twitter-openapi-typescript?

### DISCUSSION THREAD

**Discussion by darkmasta** [fa0311/fa0311#1] [General]
If I give the media_id to postCreateTweet function I can do it but is there an API for uploading the images and getting the ids? 

How can I create tweets with media easy way, correct way?

Thanks in advance.

**Comment by fa0311** [comment]
twitter-openapi-typescript does not support an API to upload an image and get its ID.
This is because the official Twitter API and the API currently used by the official client are the same.
(I don't want to redevelop something that already exists a lot.)

There is no TypeScript implementation, but here is an example in Python.
https://github.com/fa0311/twitter_openapi_python/blob/master/example/sample/tweet_media.py
Tweepy is a library for the official Twitter API, but it uploads images using tokens obtained from the web client.

I don't know if there is a good Twitter API library for TypeScript, but if I have time, I will try to make a simple sample.

**Comment by fa0311** [comment]
I implemented this using [twitter-api-v2](https://www.npmjs.com/package/twitter-api-v2).
You can do this by creating a twitter-api-v2 [Plugin](https://github.com/fa0311/twitter-openapi-typescript/blob/master/example/src/login.ts#L17-L36).

https://github.com/fa0311/twitter-openapi-typescript/blob/master/example/src/tweetMedia.ts
